agent/job_runner.go

```diff
@@ -514,6 +514,12 @@ func (r *JobRunner) createEnvironment() ([]string, error) {
 		env["BUILDKITE_AGENT_PROFILE"] = r.conf.AgentConfiguration.Profile
 	}
 
+	// PTY-mode is enabled by default in `start` and `bootstrap`, so we only need
+	// to propagate it if it's explicitly disabled.
+	if r.conf.AgentConfiguration.RunInPty == false {
+		env["BUILDKITE_PTY"] = "false"
+	}
```

Comment:
We could unconditionally set `env["BUILDKITE_PTY"]` to stringified `AgentConfiguration.RunInPty`, but I prefered to avoid introducing a new `BUILDKITE_PTY=true` environment variable to all the builds/jobs that haven't explicitly opted out of PTY mode.